### ISSUE TITLE
Remove use of platform.dist() and platform.linux_distribution() methods removed in python 3.7

### ISSUE THREAD

**Issue by nrwahl2** [ansible/ansible#30489]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
module_utils/basic.py
assorted others

##### ANSIBLE VERSION
```
ansible 2.5.0 (issue/23485 f4245e31f0) last updated 2017/09/17 18:18:48 (GMT -500)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/reid/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/reid/git/ansible/lib/ansible
  executable location = /home/reid/git/ansible/bin/ansible
  python version = 2.7.13 (default, Sep  5 2017, 08:53:59) [GCC 7.1.1 20170622 (Red Hat 7.1.1-3)]
```

##### CONFIGURATION
N/A

##### OS / ENVIRONMENT
N/A

##### SUMMARY
Several scripts use deprecated platform.dist() method. Expected to be removed in Py3.7 (https://bugs.python.org/issue1322). No plans for a standard library replacement, but nir0s/distro aims to serve the same purpose and do it better: https://github.com/nir0s/distro

We need to move away from `platform.dist()` and `platform.linux_distribution` onto either a homegrown method or nir0s/distro.

##### STEPS TO REPRODUCE
N/A

##### EXPECTED RESULTS
N/A

##### ACTUAL RESULTS
N/A

**Comment by alikins** [comment]
Sounds like a candidate for devel and backport to stable-2.4 (and maybe earlier).

**Comment by alikins** [comment]
milestone to 2.5 and flagged as p2 priority since it has to be fixed for at least 2.5 (though hopefully way before that)

**Comment by alikins** [comment]
See also https://github.com/ansible/ansible/issues/13874

https://github.com/saltstack/salt/pull/40009 is the change to salt change for this. (uses nir0s/distro)
https://github.com/pypa/pip/pull/3906 is how pip changed this. (vendors nir0s/distro)
https://github.com/nicolargo/glances/commit/af67cfcf8384204e6c8a098f34a4f4c3f0b59fc5 'glances' change (homegrown)
https://github.com/rpm-software-management/mock/commit/283c8945e3dca2aba9c756a55203bf7550855e9b  - change in rpm 'mock' (uses 'nir0s/distro')

**Comment by nrwahl2** [comment]
Thanks! bcoca and abadger suggested vendoring it in our case. I haven't had time to work on it since that discussion Sunday(?) night.

**Comment by abadger** [comment]
Note: https://github.com/nir0s/distro  is licensed Apache 2.0.  We're licensed GPLv3+.  Probably should formulate questions to ask a lawyer about using/vendoring this.  I know that GPLv3+ is compatible with Apache 2.0 but I don't know if there's anything special that they'd like us to do (for instance, noting it in documentation, keeping it in a separate file, etc).

**Comment by alikins** [comment]
Assuming bundling nir0s/distro is an option, I think that would be my preference over implementing platform.dist() analogs ourselves.

Plus, once we depend on nir0s/distro we may be able to further simplify some of our existing platform/distribution collection tools (ie, module_utils/facts/system/distribution.py etc)

**Comment by alikins** [comment]
Another option may be to bundle/vendor/include a copy of the current platform.py from the 2.7/3.6 releases. It is a relatively small amount of code for the distro stuff, so that may be easiest short term.

The license in that particular file looks BSD-ish:

```
    Copyright (c) 1999-2000, Marc-Andre Lemburg; mailto:mal@lemburg.com
    Copyright (c) 2000-2010, eGenix.com Software GmbH; mailto:info@egenix.com

    Permission to use, copy, modify, and distribute this software and its
    documentation for any purpose and without fee or royalty is hereby granted,
    provided that the above copyright notice appear in all copies and that
    both that copyright notice and this permission notice appear in
    supporting documentation or portions thereof, including modifications,
    that you make.

    EGENIX.COM SOFTWARE GMBH DISCLAIMS ALL WARRANTIES WITH REGARD TO
    THIS SOFTWARE, INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND
    FITNESS, IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL,
    INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING
    FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
    NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
    WITH THE USE OR PERFORMANCE OF THIS SOFTWARE !
```

**Comment by alikins** [comment]
https://github.com/ansible/ansible/pull/35312  is one proposal of a way to do this.

#35312 is basically 'cut & paste pythons platform.py into the tree'

Other ways to resolve this would include the previously mentioned 'distros' module or implementing it ourselves.

**Comment by tirkarthi** [comment]
`platform.linux_distribution` was removed in CPython master that will be released with Python 3.8 . Ref : https://bugs.python.org/issue28167

Thanks
